frontend/javascripts/oxalis/controller/combinations/segmentation_handlers.js

```diff
@@ -4,45 +4,72 @@ import Model from "oxalis/model";
 import { calculateGlobalPos } from "oxalis/model/accessors/view_mode_accessor";
 import { getMappingInfo } from "oxalis/model/accessors/dataset_accessor";
 import { loadAgglomerateSkeletonAction } from "oxalis/model/actions/skeletontracing_actions";
-import Store from "oxalis/store";
+import Store, { type OxalisState } from "oxalis/store";
 import Toast from "libs/toast";
 import messages from "messages";
 import { clickSegmentAction } from "oxalis/model/actions/volumetracing_actions";
 import api from "oxalis/api/internal_api";
 import { getSegmentIdForPosition } from "oxalis/controller/combinations/volume_handlers";
 
+export function hasAgglomerateMapping(state: OxalisState) {
+  const segmentation = Model.getVisibleSegmentationLayer();
+  if (!segmentation) {
+    return {
+      value: false,
+      reason: "",
+    };
+  }
+  const { mappingName, mappingType, mappingStatus } = getMappingInfo(
+    state.temporaryConfiguration.activeMappingByLayer,
+    segmentation.name,
+  );
+  if (mappingName == null || mappingStatus !== MappingStatusEnum.ENABLED) {
+    return {
+      value: false,
+      reason: messages["tracing.agglomerate_skeleton.no_mapping"],
+    };
+  }
+
+  if (mappingType !== "HDF5") {
+    return {
+      value: false,
+      reason: messages["tracing.agglomerate_skeleton.no_agglomerate_file"],
+    };
+  }
+  return {
+    value: true,
+    reason: "",
+  };
+}
+
 export async function handleAgglomerateSkeletonAtClick(clickPosition: Point2) {
```

Comment:
Can you check whether this function is still used? If not, it can be removed.

Reply:
Seems like the middle click was the only place this was used, so unless that functionality is supposed to make a return I will delete the method

Reply:
Exactly, it can be removed :+1: